### ISSUE TITLE
[REQ] Block accounts / Blacklist / Filter tweets

### ISSUE THREAD

**Issue by Mrnofish** [jonjomckay/fritter#555]
**Describe the problem that needs solving**

Some accounts I follow are plagued by haters that spew their unpleasant juice all over, making the comment stream more difficult to parse quickly.

**Describe the solution you'd like**
When viewing a tweet, there could be a "Block User" in the three-dots menu. This would add the account to a blacklist, and either hide their tweets in comments, or replace the tweets with a small "this user has been blocked by you" pane.

Since sometimes haters can be identified by some specific words they use, it could also be a good idea to implement filtering based on words or with perhaps optional regex support.

The latter solution could complement the first, making it easier, in certain scenarios, to block unwanted content without having to blacklist many different users manually, which can become a time-consuming task.

OTOH filters are less targeted than account blocking, and might occasionally cause issues with content being hidden inadvertently, so account blocking is IMO more essential, and filtering is more of a luxury. If filtering were implemented, it would also be very nice to have an extra option to quickly show a tweet that's been blocked by filters.

I appreciate this might take quite some work to implement, thanks for the app and taking this FR into consideration.
